java-api-core/src/com/kinvey/java/store/BaseDataStore.java

```diff
@@ -536,6 +537,7 @@ private KinveyPullResponse pullBlockingPaged(@Nonnull Query query, int pageSize)
             }
             executor.shutdown();
         }
+        query.setSkip(0).setLimit(0); // To set back default value of skip and limit
```

Comment:
What is the reason for setting the default values on the `skip` and `limit` modifiers here?

Reply:
How I wrote above, the SDK sets own `skip` and `limit` for auto-pagination, and after pagination pull user's query will have `skip` and `limit` from the last page of pagination request even if the user didn't set `skip` or `limit` before calling `pull` with auto-pagination. We can exclude this from the PR, but I think it's more user-friendly.

Reply:
That's fine. Another potential way of doing this in the future would be to clone the user's query, and apply `skip` and `limit` modifiers on the clone for AP purposes. That way, the user's query remains untouched.